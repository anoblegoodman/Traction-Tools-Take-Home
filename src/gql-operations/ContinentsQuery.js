import { gql } from 'apollo-boost';

export const CONTINENTS_QUERY = gql`
  {
    continents {
      code
      name
    }
  }
`;
