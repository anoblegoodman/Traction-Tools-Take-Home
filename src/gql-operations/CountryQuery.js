import { gql } from 'apollo-boost';

export const COUNTRY_QUERY = gql`
  query Continent($code: ID!) {
    continent(code: $code) {
      code
      name
      countries {
        name
        emoji
        languages {
          name
        }
      }
    }
  }
`;
