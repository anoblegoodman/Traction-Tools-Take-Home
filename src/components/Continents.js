import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import { Route } from 'react-router-dom';
import { CONTINENTS_QUERY } from '../gql-operations/ContinentsQuery.js';
import {
  ContinentsButton,
  Header,
  LoadingIcon,
  ListWrapper
} from '../styles/styles';

export const Continents = () => {
  const { loading, error, data } = useQuery(CONTINENTS_QUERY);

  if (loading) return <LoadingIcon size="300px" />;
  if (error) return <p>Error</p>;
  if (!data) return null;

  return (
    <div>
      <Header>Choose your continent</Header>
      <ListWrapper>
        {data.continents.map((continent, i) => (
          <Route
            key={`continent${i}`}
            children={({ history }) => {
              return (
                <ContinentsButton
                  data-testid={`${continent.name}`}
                  onClick={() =>
                    history.push({
                      pathname: `${history.location.pathname}/${continent.name}`,
                      state: { continentCode: continent.code }
                    })
                  }
                >{`${continent.name}`}</ContinentsButton>
              );
            }}
          ></Route>
        ))}
      </ListWrapper>
    </div>
  );
};
