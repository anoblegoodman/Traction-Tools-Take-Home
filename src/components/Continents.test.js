import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import wait from 'waait';
import { MockedProvider } from '@apollo/react-testing';
import TestRenderer from 'react-test-renderer';
import { CONTINENTS_QUERY } from '../gql-operations/ContinentsQuery';
import { Continents } from './Continents';

const defaultMocks = [
  {
    request: {
      query: CONTINENTS_QUERY
    },
    result: {
      data: {
        continents: [
          {
            code: 'AF',
            name: 'Africa'
          }
        ]
      }
    }
  }
];

describe('<Continents />', () => {
  it('should match snapshot when there are continents', async () => {
    const json = TestRenderer.create(
      <MemoryRouter>
        <MockedProvider addTypename={false} mocks={defaultMocks}>
          <Continents />
        </MockedProvider>
      </MemoryRouter>
    );
    await wait(1);
    await expect(json).toMatchSnapshot();
  });
});
