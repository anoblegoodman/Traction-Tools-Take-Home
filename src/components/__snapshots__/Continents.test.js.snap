// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Continents /> should match snapshot when there are continents 1`] = `
<div>
  <div
    className="sc-AxheI gPSmPT"
  >
    Choose your continent
  </div>
  <div
    className="sc-fznyAO jvkATr"
  >
    <button
      className="sc-AxgMl eMOtsN"
      data-testid="Africa"
      onClick={[Function]}
    >
      Africa
    </button>
  </div>
</div>
`;
